### ISSUE TITLE
Applying database changes failed

### ISSUE THREAD

**Issue by Mek0n21** [orangehrm/orangehrm#1783]
### OrangeHRM version

5.5

### Environment details

- OrangeHRM source:
- Platform:
- PHP version:
- Database and version:
- Web server:
- Browser:

### Describe the bug

Failing to install stops aat 50%  - Request failed with status code 500. please help

### To reproduce

Install

### Expected behavior

Installation to be successful

### Relevant log output

```shell
[2024-01-08T16:54:22.059582+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_clean_up_install","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\CleanUpInstallAPI::handle","_route":"api_installer_clean_up_install"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/clean-up-install","method":"POST"} []
[2024-01-08T16:54:22.120516+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:22.448201+00:00] installer.INFO: Matched route "{route}". {"route":"installer_confirmation","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\ConfirmationController::handle","_route":"installer_confirmation"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/confirmation","method":"GET"} []
[2024-01-08T16:54:22.683518+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_database_config","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\DatabaseConfigAPI::handle","_route":"api_installer_database_config"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/database-config","method":"GET"} []
[2024-01-08T16:54:22.717050+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_admin_user","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\AdminUserAPI::handle","_route":"api_installer_admin_user"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/admin-user","method":"GET"} []
[2024-01-08T16:54:28.624633+00:00] installer.INFO: Matched route "{route}". {"route":"installer_process","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\InstallerController::handle","_route":"installer_process"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/process","method":"GET"} []
[2024-01-08T16:54:28.789560+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_start_registration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\InstallerDataRegistrationAPI::handle","_route":"api_installer_start_registration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/send-data/installer-start","method":"POST"} []
[2024-01-08T16:54:29.147811+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_database","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\InstallationDatabaseAPI::handle","_route":"api_installer_database"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/database","method":"POST"} []
[2024-01-08T16:54:29.152402+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:29.185372+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_pre_migration_check","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\PreMigrationCheck::handle","_route":"api_installer_pre_migration_check"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/pre-migration","method":"POST"} []
[2024-01-08T16:54:29.192161+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:29.326152+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_versions","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\VersionAPI::handle","_route":"api_installer_versions"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/versions?excludeLatest=false","method":"GET"} []
[2024-01-08T16:54:29.358344+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:29.360294+00:00] installer.INFO: {"version":"3.3.3"} [] []
[2024-01-08T16:54:29.364995+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:30.955478+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:30.956286+00:00] installer.INFO: {"version":"4.0"} [] []
[2024-01-08T16:54:30.960046+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.028477+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.029561+00:00] installer.INFO: {"version":"4.1"} [] []
[2024-01-08T16:54:31.033562+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.133734+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.134603+00:00] installer.INFO: {"version":"4.1.1"} [] []
[2024-01-08T16:54:31.138356+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.233884+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.234778+00:00] installer.INFO: {"version":"4.1.2"} [] []
[2024-01-08T16:54:31.238551+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.283938+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.284730+00:00] installer.INFO: {"version":"4.1.2.1"} [] []
[2024-01-08T16:54:31.287906+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.363391+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.364165+00:00] installer.INFO: {"version":"4.2"} [] []
[2024-01-08T16:54:31.367893+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.446478+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.447267+00:00] installer.INFO: {"version":"4.2.0.1"} [] []
[2024-01-08T16:54:31.450646+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.501786+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.502539+00:00] installer.INFO: {"version":"4.3"} [] []
[2024-01-08T16:54:31.506337+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.583169+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.583962+00:00] installer.INFO: {"version":"4.3.1"} [] []
[2024-01-08T16:54:31.587274+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.644520+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.645264+00:00] installer.INFO: {"version":"4.3.2"} [] []
[2024-01-08T16:54:31.648414+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.708630+00:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"https://proactiveportal.co.uk/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2024-01-08T16:54:31.709372+00:00] installer.INFO: {"version":"4.3.3"} [] []
[2024-01-08T16:54:31.712797+00:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"4.18.0-477.13.1.lve.el8.x86_64","version_info":"#1 SMP Thu Jun 1 16:40:47 EDT 2023"},"php":{"version":"8.2.13"},"mysql":{"client_version":"3.3.8","server_version":"10.6.16-MariaDB-cll-lve","conn_type":"Localhost via UNIX socket"},"server":"Apache","ohrm":{"version":"5.5"}} [] []
[2024-01-08T16:54:31.760427+00:00] installer.ERROR: An exception occurred while executing a query: SQLSTATE[HY000]: General error: 1833 Cannot change column 'id': used in a foreign key constraint 'ohrm_job_interview_ibfk_1' of table 'proactiveportal_orange_hrm.ohrm_job_interview' [] []
[2024-01-08T16:54:31.760559+00:00] installer.ERROR: #0 /home/proactiveportal/public_html/src/vendor/doctrine/dbal/src/Connection.php(1814): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert() #1 /home/proactiveportal/public_html/src/vendor/doctrine/dbal/src/Connection.php(1749): Doctrine\DBAL\Connection->handleDriverException() #2 /home/proactiveportal/public_html/src/vendor/doctrine/dbal/src/Connection.php(1163): Doctrine\DBAL\Connection->convertExceptionDuringQuery() #3 /home/proactiveportal/public_html/src/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(1204): Doctrine\DBAL\Connection->executeStatement() #4 /home/proactiveportal/public_html/src/vendor/doctrine/dbal/src/Schema/AbstractSchemaManager.php(820): Doctrine\DBAL\Schema\AbstractSchemaManager->_execSql() #5 /home/proactiveportal/public_html/installer/Util/V1/SchemaHelper.php(153): Doctrine\DBAL\Schema\AbstractSchemaManager->alterTable() #6 /home/proactiveportal/public_html/installer/Migration/V4_3_3/Migration.php(54): OrangeHRM\Installer\Util\V1\SchemaHelper->changeColumn() #7 /home/proactiveportal/public_html/installer/Util/AppSetupUtility.php(526): OrangeHRM\Installer\Migration\V4_3_3\Migration->up() #8 /home/proactiveportal/public_html/installer/Util/AppSetupUtility.php(514): OrangeHRM\Installer\Util\AppSetupUtility->_runMigration() #9 /home/proactiveportal/public_html/installer/Controller/Upgrader/Api/MigrationAPI.php(50): OrangeHRM\Installer\Util\AppSetupUtility->runMigrationFor() #10 /home/proactiveportal/public_html/installer/Controller/AbstractInstallerRestController.php(57): OrangeHRM\Installer\Controller\Upgrader\Api\MigrationAPI->handlePost() #11 /home/proactiveportal/public_html/installer/Controller/AbstractInstallerController.php(85): OrangeHRM\Installer\Controller\AbstractInstallerRestController->execute() #12 /home/proactiveportal/public_html/src/vendor/symfony/http-kernel/HttpKernel.php(152): OrangeHRM\Installer\Controller\AbstractInstallerController->handle() #13 /home/proactiveportal/public_html/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #14 /home/proactiveportal/public_html/installer/Framework/HttpKernel.php(244): Symfony\Component\HttpKernel\HttpKernel->handle() #15 /home/proactiveportal/public_html/installer/index.php(66): OrangeHRM\Installer\Framework\HttpKernel->handleRequest() #16 {main} [] []
```

**Comment by devishke-orange** [comment]
Hi @Mek0n21 ,

This is a known issue. Please see https://github.com/orangehrm/orangehrm/issues/1731#issuecomment-1727583099

**Comment by devishke-orange** [comment]
Hi @Mek0n21,

This issue is fixed in OrangeHRM 5.6: https://github.com/orangehrm/orangehrm/releases/tag/v5.6

**Comment by devishke-orange** [comment]
I'm closing this since the issue was fixed.
